### PR TITLE
PR: Fix issue #3436 

### LINKED ISSUE #3436: Default file extension should be .py instead of empty
## The default file type is empty in the save dialogs

**What steps will reproduce the problem?**

1. Open any document and use save as dialog
2. if there exists any remove the extension
3. Click save

**What is the expected output? What do you see instead?**

The expected outcome (I think) the file should be save with a defauly .py extension instead of no-extension state. 

**Please provide any additional information below**

You can see from the screenshot possible extension recommendation windows is also empty

zzzz

## Versions and main components

* Spyder Version: 3.0.0b7
* Python Version: 3.6
* Operating system: Win10

## Dependencies
jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.1 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.1 (OK)
pandas >=0.13.1  :  0.18.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.3.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.4.6 (OK)
sympy >=0.7.3    :  None (NOK)

### PR DESCRIPTION
**Pull request by malliwi88**
Fixes #3436

Default file extension should be .py (Save_as Widget).

### CODE REVIEW

**Comment by malliwi88** [comment]
@goanpeca I have a Read-only Zenhub Board, please. You can change the Milestone for this PR

**Comment by goanpeca** [comment]
@malliwi88 great :-), this is exactly the fix I had in mind ;-)

File: spyder/widgets/editor.py [hunk 1]
```diff
@@ -1302,7 +1304,11 @@ def file_saved_in_other_editorstack(self, index, filename):
     def select_savename(self, original_filename):
         self.redirect_stdio.emit(False)
         filename, _selfilter = getsavefilename(self, _("Save file"),
-                                               original_filename)
+                                               original_filename,
+                                               get_edit_filters(),
+                                               get_filter(get_edit_filetypes(),
+                                                          osp.splitext(original_filename)[1])
+                                               )
```
Comment (goanpeca) [inline spyder/widgets/editor.py]: what happens if the file does not have an extension?

Could you test if you open a file without extension on the editor?

for example this one https://github.com/spyder-ide/spyder/blob/master/LICENSE

Comment (malliwi88) [inline spyder/widgets/editor.py]: Hi @goanpeca these are my results just look like that on:

* Win 7 
fix-save_as win 7 seleccion_006

* Linux
fix-save_as seleccion_006

**Comment by goanpeca** [comment]
Ok LGTM :-)

File: spyder/widgets/editor.py [hunk 2]
```diff
@@ -1302,7 +1304,11 @@ def file_saved_in_other_editorstack(self, index, filename):
     def select_savename(self, original_filename):
         self.redirect_stdio.emit(False)
         filename, _selfilter = getsavefilename(self, _("Save file"),
-                                               original_filename)
+                                               original_filename,
+                                               get_edit_filters(),
+                                               get_filter(get_edit_filetypes(),
```
Comment (ccordoba12) [inline spyder/widgets/editor.py]: Perhaps if we change here `get_edit_filetypes` to `EDIT_FILETYPES` we could get the expected result :-)

**Pull request merged**